### DISCUSSION TITLE
Using fetched data inside of methods?

### DISCUSSION THREAD

**Discussion by AlexDicy** [nuxt/nuxt#18060] [Questions] [answered]
Example
```html
<script setup>
const { data } = await useFetch('/api/count')
</script>

<template>
  Page visits: {{ data.count }}
</template>
```

accessing `data` works because the template is rendered inside the `setup` function when building the server using `nuxi build`.
this variable is not stored in the component's context, so it cannot be used in a component's method

```html
<script>
export default {
  methods: {
    getLink() {
      return `https://website/product/${this.data}`;
    }
  }
};
</script>
```

this works fine when running `npm dev` but breaks when using the server from `nuxi build`, as `data` is null.

Is there any workaround/correct way todo this?

**Comment by lucpotage** [comment, marked as answer]
Did you try something like this?
```vue
<script setup>
const { data } = await useFetch('/api/count')
const getLink = () => `https://website/product/${data.value}`
</script>
```

**Reply by wobsoriano** [reply]
or better with a computed property?

```js
const getLink = () => computed(() => `https://website/product/${data.value}`)
```

**Reply by AlexDicy** [reply]
unfortunately, even after importing `{computed} from "vue"` or from `#imports` using `computed` returns nothing when used both as `{{ getLink }}` and `{{ getLink() }}`

**Reply by danielroe** [reply]
I suspect @wobsoriano meant:

```js
const link = computed(() => `https://website/product/${data.value}`)
```

... and then use `link` in your template.

**Reply by AlexDicy** [reply]
it does indeed work that way
